.changeset/tangy-nails-hide.md

```diff
@@ -0,0 +1,10 @@
+---
+"@pnpm/plugin-commands-installation": minor
+"@pnpm/resolve-dependencies": minor
+"@pnpm/workspace.manifest-writer": minor
+"@pnpm/core": minor
+"@pnpm/config": minor
+"pnpm": minor
+---
+
+Add a CLI flag named `--save-catalog` to `pnpm add` to save new dependencies as a catalog: `catalog:` will be added to `package.json` and the package specifier will be added to the default catalog object (either `catalog` or `catalogs.default`) in `pnpm-workspace.yaml` [#9425](https://github.com/pnpm/pnpm/issues/9425).
```

Comment:
should we make it possible to pass a name? So `--save-catalog=react` would add it to the named catalog called "react".

Reply:
I tried `'save-catalog': [Boolean, String],` in `types.ts` but `pd add --save-catalog pkg-name` would consume `pkg-name`.

Reply:
What about [String, Boolean] ?

There is one flag that uses a similar type: `color: ['always', Boolean],`

Reply:
If it won't work, just make it a string and I'll make a workaround to make it work with `--save-catalog`.

Reply:
`[String, Boolean]` doesn't work either.

> If it won't work, just make it a string and I'll make a workaround to make it work with `--save-catalog`.

I have noticed that pnpm uses `@pnpm/nopt` as a parser for CLI arguments.

Reply:
Agree with allowing this to accept a name if that's possible. 👍